app/src/tunnel/kotlin/tunnel/Blockade.kt

```diff
@@ -36,31 +51,90 @@ internal class Blockade(
+            val arraylist = ruleset.toString().split(",")//", ", "[", "]")
+            ruleset.removeAll(whitelistRuleset)
+            ruleset.removeAll(wildcardRuleset)
+            if (octomize == true) {
+                for (item in arraylist) {
+                    var iit = item
+                    iit = iit.removePrefix("[")
+                    iit = iit.removeSuffix("]")
+                    iit = iit.trim()
+                    counter++
+                    if (inwildcardlist(iit)) {
+                        ruleset.remove(iit)
+                        // makes process much faster when using Log.d on large list to only see one out of 2500 host
+                        if (counter%1000==1)
+                        // detailed Log.d version
+                        // android.util.Log.d("skipped", "number of host left to evaluate: " + (orignalrulesize-counter)+ " orignal number of host: " + orignalrulesize + " octomized number of host in list: " + ruleset.size + "  percent finished is: " +((counter*100)/orignalrulesize) +  "% percent saved is: " + (((orignalrulesize - ruleset.size)*100)/counter)+ "% " + orignalrulesize +" " + ruleset.size + " " + counter + " host name: " + iit + " memory is " + Runtime.getRuntime().freeMemory() + " "  )
+                        // easy view Log.d version
+                         android.util.Log.d("skipped", ""+ (orignalrulesize-counter)+ " " + orignalrulesize + " " + ruleset.size + "  percent finished is " +((counter*100)/orignalrulesize) +  "% percent saved is " + (((orignalrulesize - ruleset.size)*100)/counter)+ "% " + orignalrulesize +" " + ruleset.size + " " + counter + " " + iit + " memory is " + Runtime.getRuntime().freeMemory() + " "  )
+                    } else {
+                        // makes process much faster when using Log.d on large list to only see one out of 2500 host
+                        if (counter%1000==2500)
+                        // detailed Log.d version
+                        // android.util.Log.d("allowed", "number of host left to evaluate: " + (orignalrulesize-counter)+ " orignal number of host: " + orignalrulesize + " octomized number of host in list: " + ruleset.size + "  percent finished is: " +((counter*100)/orignalrulesize) +  "% percent saved is: " + (((orignalrulesize - ruleset.size)*100)/counter)+ "% " + orignalrulesize +" " + ruleset.size + " " + counter + " host name: " + iit + " memory is " + Runtime.getRuntime().freeMemory() + " "  )
+                        // easy view Log.d version
+                         android.util.Log.d("allowed", ""+ (orignalrulesize-counter)+ " " + orignalrulesize + " " + ruleset.size + "  percent finished is " +((counter*100)/orignalrulesize) +  "% percent saved is " + (((orignalrulesize - ruleset.size)*100)/counter)+ "% " + orignalrulesize +" " + ruleset.size + " " + counter + " " + iit + " memory is " + Runtime.getRuntime().freeMemory() + " "  )
```

Comment:
This will never be called because `counter%1000` is always < 1000.

It would be cleaner to remove the `else` around this log (in line 90) and remove the log in line 89 completely.